packages/lib-classifier/src/components/Classifier/components/SubjectViewer/helpers/getDataSeriesColor/getDataSeriesColor.spec.js

```diff
@@ -0,0 +1,118 @@
+import sinon from 'sinon'
+import zooTheme from '@zooniverse/grommet-theme'
+import getDataSeriesColor from './getDataSeriesColor'
+import variableStar from '../mockLightCurves/variableStar'
+import {
+  dataSeriesWithCustomColor,
+  dataSeriesWithInvalidCustomColor,
+  dataSeriesWithThemeColorVariables
+} from './helpers/mocks'
+
+const defaultColors = Object.values(zooTheme.global.colors.drawingTools)
+
+describe('Helper > getDataSeriesColor', function () {
+  it('should be a function', function () {
+    expect(getDataSeriesColor).to.be.a('function')
+  })
+
+  it('should default to return an empty string', function () {
+    const color = getDataSeriesColor()
+    expect(color).to.be.a('string')
+    expect(color).to.be.empty()
+  })
+
+  describe('with data series options', function () {
+    it('should return the color from the theme if using theme variable name', function () {
+      dataSeriesWithThemeColorVariables.data.forEach((series, seriesIndex) => {
+        const color = getDataSeriesColor({
+          seriesOptions: series.seriesOptions,
+          seriesIndex,
+          themeColors: zooTheme.global.colors
+        })
+        expect(color).to.equal(zooTheme.global.colors[series.seriesOptions.color])
+      })
+    })
+
+    it('should return the color defined in the subject JSON for each data series\' options object', function () {
+      dataSeriesWithCustomColor.data.forEach((series) => {
+        const color = getDataSeriesColor({ seriesOptions: series.seriesOptions })
+        expect(color).to.equal(series.seriesOptions.color)
+      })
+    })
+
+    describe('when the color is invalid', function () {
+      let logError
+      before(function () {
+        logError = sinon.stub(console, 'error')
+      })
+
+      after(function () {
+        console.error.restore()
+      })
+
+      it('should error to the console and return an empty string', function () {
+        dataSeriesWithInvalidCustomColor.data.forEach((series) => {
```

Comment:
Can you help me understand how an invalid color could be submitted and why we would want to return a blank string (`''`) instead of the default `light-4` here? Is this mainly just to cover every potential edge case?

Reply:
We're allowing researchers to specify a color in their subject JSON because color might have meaning in the research, like with the variable star use case, where red and blue represent specific spectra.

I've also chosen to support specifying colors to use by way of a theme object, like our zooniverse Grommet theme, but it can possibly be changed out for some other theme object that uses the same structure and conventions. It's possible this can also have an error in it. So instead of a fallback color in these edge cases, the data points render without a fill color and the console has the error message. Seeing the data points without a fill color should hopefully give the researcher or the dev a clue that something is wrong.

Reply:
This makes sense. Good call on not displaying the error so the researcher can respond accordingly.